src/Controller/BookController.php

```diff
@@ -29,13 +36,79 @@ public function index(): Response
 
   public function searchBooksByTitle(string $title): Response
   {
+    $title = urldecode($title);
+    $bookRepository = $this->entityManager->getRepository(Book::class);
+    // $book = $bookRepository->findOneBy(['title' => $title]);
+    $books = $bookRepository->createQueryBuilder('b')
+    ->where('b.title LIKE :title')
+    ->setParameter('title', '%' . $title . '%')
```

Comment:
add the urldecode($title) here:
`->setParameter('title', '%' . $title . '%')`

the urldecode($title) will work for our database, but not for the api, if the title contains spaces. 
With my suggested change both will work

Reply:
done!